File: java/ankit_kanyal/devnest_problems/StreetParade.java
// https://devsnest.in/algo-challenges/street-parade

static int solve(int n, int[] order){
//CODE HERE 
Stack<Integer>st = new Stack<>();
int min = 0;

int flag = 0;

for(int i = 0;i<n-1;i++){

while(st.size()>0 && st.peek()<order[i]){

        st.pop();

    }

if(order[i]>order[i+1]){

    st.push(order[i]);

}
else if(order[i]<min){

    flag = 1;

    break;

}
else{

    min = order[i];

}
}

if(flag==1){

return 0;
}

else{

return 1;
} 
}

File: Python/vedant_bhatnagar/DevsnestProblems_THA/Phase 1/streetparade.py
def solve(n, order):
    # CODE HERE
    st = []
    ans = []
    for i in order:
        if not st:
            st.append(i)
            continue
        while st and st[-1]<i:
            ans.append(st.pop())
        st.append(i)
    
    if st:
        while st:
            ans.append(st.pop())
    # print(ans)
    for i in range(0, len(order)-1):
        if ans[i] > ans[i+1]:
            return 0
    return 1
        